x-pack/plugin/sql/sql-action/src/main/java/org/elasticsearch/xpack/sql/action/SqlClearCursorRequest.java

```diff
@@ -20,30 +19,38 @@
 
 import static org.elasticsearch.action.ValidateActions.addValidationError;
 import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+import static org.elasticsearch.xpack.sql.action.AbstractSqlQueryRequest.CURSOR;
+import static org.elasticsearch.xpack.sql.action.AbstractSqlQueryRequest.MODE;
+import static org.elasticsearch.xpack.sql.action.AbstractSqlQueryRequest.CLIENT_ID;
 
 /**
  * Request to clean all SQL resources associated with the cursor
  */
 public class SqlClearCursorRequest extends AbstractSqlRequest {
 
-    private static final ConstructingObjectParser<SqlClearCursorRequest, Mode> PARSER =
-        new ConstructingObjectParser<>(SqlClearCursorAction.NAME, true, (objects, mode) -> new SqlClearCursorRequest(
-            mode,
-            (String) objects[0]
-        ));
+    private static final ConstructingObjectParser<SqlClearCursorRequest, RequestInfo> PARSER =
+        // here the position in "objects" is the same as the fields parser declarations below 
+        new ConstructingObjectParser<>(SqlClearCursorAction.NAME, objects -> {
+            RequestInfo requestInfo = new RequestInfo(Mode.fromString((String) objects[1]),
+                    (String) objects[2]);
+            return new SqlClearCursorRequest(requestInfo, (String) objects[0]);
+       });
 
     static {
-        PARSER.declareString(constructorArg(), new ParseField("cursor"));
+        // "cursor" is required constructor parameter
+        PARSER.declareString(constructorArg(), CURSOR);
+        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), MODE);
+        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), CLIENT_ID);
     }
 
     private String cursor;
 
     public SqlClearCursorRequest() {
-
+        super();
```

Comment:
There's no value in adding `super` as this constructor is called implicitly anyway.